CMakeLists.txt

```diff
@@ -2,93 +2,173 @@ cmake_minimum_required (VERSION 3.5)
 
 project (korc_proj C CXX Fortran)
 
-#  Set directories to build all binary files.
-set (EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR}/build/bin)
-
-#  Fortran specific settings.
-set (CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -cpp")
-set (CMAKE_Fortran_MODULE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/build/modules)
-set (CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/build/lib)	
+#  FIXME: Remove once link issues are resolved with the new macos linker.
+if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+   SET (CMAKE_EXE_LINKER_FLAGS -ld_classic)
+endif ()
 
-#  Set definitions.
-add_definitions (-DDOUBLE_PRECISION)
-add_definitions (-DPARALLEL_RANDOM)
-if (CMAKE_BUILD_TYPE STREQUAL "Debug")
-#   add_definitions (-DDBG_CHECK)
-endif()
+#-------------------------------------------------------------------------------
+#  Setup build types.
+#-------------------------------------------------------------------------------
+set_property (CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS
+              Debug
+              Release
+)
 
-#  Configure OpenMP
-find_package (OpenMP)
-set (CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} ${OpenMP_Fortran_FLAGS}")
-set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_Fortran_FLAGS}")
-option (USE_OMP "Use OpenOMP" OFF)
-if (${USE_OMP})
-    add_definitions (-DOMP)
-endif()
+#  Make sure the prefix path appears as a ccmake variable.
+set (CMAKE_PREFIX_PATH "" CACHE PATH "Search paths for dependency look up.")
```

Comment:
Why would a developers or users want to modify the `CMAKE_PREFIX_PATH` in a specific build? 

If we really need this, I suspect it could be done with [mark_as_advanced](https://cmake.org/cmake/help/latest/command/mark_as_advanced.html)

Reply:
The find_package commands can fail when libraries are installed in non standard locations. `CMAKE_PREFIX_PATH` isn't exposed from ccmake. This allows you to set the prefix path without leaving ccmake. No mater what I set for `mark_as_advanced` it doesn't show up any where.

Reply:
I'm trying to better understand the underlying need. Our two external dependencies (HDF5 and MPI) have find_package modules that work just work much of the time and offer `HDF5_DIR` and `MPI_DIR` variables for additional hints.

Reply:
HDF5_DIR isn't listed as an hint in the documentation. https://cmake.org/cmake/help/latest/module/FindHDF5.html The current version of FindMPI has an `MPI_HOME` variable https://cmake.org/cmake/help/v3.28/module/FindMPI.html but it's not in every version. I would be hesitant relying on variables that are not documented.

Reply:
Oops, try `HDF5_ROOT` or `MPI_ROOT`
https://cmake.org/cmake/help/latest/variable/PackageName_ROOT.html